modules/govuk_ci/templates/usr/local/bin/remove_old_jenkins_jobs.sh.erb

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+key_str=<%= @pipeline_jobs.keys.join("|") %>
+ls -a /var/lib/jenkins/jobs | grep -vE "$key_str|\.|\.\." | xargs rm -rf
```

Comment:
I'd recommend using the `-F` fixed string option to grep (takes a newline-separated list of fixed strings to match against), to avoid potential issues with quoting. Also `ls -A` excludes the current and parent dir for you so you can simplify the grep a bit. So something like

```suggestion
dirs_to_keep='<%= @pipeline_jobs.keys.join("\n") %>'

ls -A /var/lib/jenkins/jobs | grep -vF "$dirs_to_keep" | xargs rm -rf
```

Reply:
(I'm assuming you want to delete directories that start with `.`. If not then just omit the `-A` altogether of course.)

Reply:
Nice 👍